### ISSUE TITLE
Unable to rebuild the plugin using CLion's Platform SDK

### ISSUE THREAD

**Issue by agatti** [Darkyenus/glsl4idea#23]
As per subject, I've tried to rebuild the plugin from source following the given instructions, but it seems like CLion's Platform SDK is missing a few things that this plugin needs.

Here's the compilation report (with the latest version from the repo):

<pre>
glsl4idea/src/glslplugin/components/GLSLCompiler.java
Error:(24, 37) java: package com.intellij.openapi.compiler does not exist
Error:(25, 37) java: package com.intellij.openapi.compiler does not exist
Error:(26, 37) java: package com.intellij.openapi.compiler does not exist
Error:(27, 37) java: package com.intellij.openapi.compiler does not exist
Error:(39, 38) java: cannot find symbol
  symbol: class TranslatingCompiler
Error:(46, 42) java: cannot find symbol
  symbol:   class CompileScope
  location: class glslplugin.components.GLSLCompiler
Error:(52, 62) java: cannot find symbol
  symbol:   class CompileContext
  location: class glslplugin.components.GLSLCompiler
Error:(57, 25) java: cannot find symbol
  symbol:   class CompileContext
  location: class glslplugin.components.GLSLCompiler
Error:(57, 97) java: cannot find symbol
  symbol:   class OutputSink
  location: class glslplugin.components.GLSLCompiler
Error:(86, 42) java: cannot find symbol
  symbol:   class OutputItem
  location: class glslplugin.components.GLSLCompiler
Error:(76, 16) java: cannot find symbol
  symbol:   class OutputItem
  location: class glslplugin.components.GLSLCompiler.ExitResult
Error:(107, 17) java: cannot find symbol
  symbol:   class CompileContext
  location: class glslplugin.components.GLSLCompiler.GLSLCompileAction
Error:(110, 34) java: cannot find symbol
  symbol:   class CompileContext
  location: class glslplugin.components.GLSLCompiler.GLSLCompileAction
Error:(45, 5) java: method does not override or implement a method from a supertype
Error:(51, 5) java: method does not override or implement a method from a supertype
Error:(56, 5) java: method does not override or implement a method from a supertype
Error:(78, 24) java: cannot find symbol
  symbol:   class OutputItem
  location: class glslplugin.components.GLSLCompiler.ExitResult
Error:(122, 36) java: cannot find symbol
  symbol:   variable CompilerMessageCategory
  location: class glslplugin.components.GLSLCompiler.GLSLCompileAction
glsl4idea/src/glslplugin/actions/GLSLDeduceExpressionTypeAction.java
Error:(24, 41) java: cannot find symbol
  symbol:   class DataKeys
  location: package com.intellij.openapi.actionSystem
Error:(33, 29) java: cannot find symbol
  symbol:   class PsiUtil
  location: package com.intellij.psi.util
Error:(80, 41) java: cannot find symbol
  symbol:   variable DataKeys
  location: class glslplugin.actions.GLSLDeduceExpressionTypeAction
Error:(119, 40) java: cannot find symbol
  symbol:   variable DataKeys
  location: class glslplugin.actions.GLSLDeduceExpressionTypeAction
Error:(120, 41) java: cannot find symbol
  symbol:   variable DataKeys
  location: class glslplugin.actions.GLSLDeduceExpressionTypeAction
Error:(129, 26) java: cannot find symbol
  symbol:   variable PsiUtil
  location: class glslplugin.actions.GLSLDeduceExpressionTypeAction
glsl4idea/src/glslplugin/structureview/GLSLStructureViewFactory.java
Error:(34, 52) java: <anonymous glslplugin.structureview.GLSLStructureViewFactory$1> is not abstract and does not override abstract method createStructureViewModel(com.intellij.openapi.editor.Editor) in com.intellij.ide.structureView.TreeBasedStructureViewBuilder
</pre>

I'll see if I can fix some of those, hopefully without breaking the plugin in IDEA.

**Comment by agatti** [comment]
TL;DR: What was deprecated on IDEA 14.1 has been removed in CLion - this means that the next IntelliJ IDEA point version will break the plugin.

I've fixed the compilation errors here: https://github.com/agatti/glsl4idea/tree/clion-1.0 - however I wasn't able to make CLion pick the plugin up.

Still, I've tried to build the plugin using IntelliJ IDEA Ultimate 14.1 and all but one errors I get in CLion are related to deprecated functionalities in the IDEA SDK.  Moreover, GLSLStructureViewFactory doesn't compile on IDEA either, you may want to see whether my modifications make it work there.

**Comment by Darkyenus** [comment]
I have manually merged your changes and removed remaining deprecated parts, can you please try compiling it now?

**Comment by agatti** [comment]
After I replace all calls to FileTemplateManager.getInstance() with FileTemplateManager.getDefaultInstance() it builds without warnings or errors using the CLion SDK.  It still doesn't load for some other reason though, as it seems like it's not activated or something.  As you can see in the attached image, it gets disabled by CLion but providing no meaningful error message in the console when launching it:

![Plugin pane](https://cloud.githubusercontent.com/assets/235740/7221087/45e682f0-e6de-11e4-9dc8-7b260e689b25.png)

I'll try packaging the plugin into a JAR myself and see if things change if installed that way.

**Comment by agatti** [comment]
Same thing, I get this in the log:

```2015-04-19 21:57:12,189 [   1268]   INFO - llij.ide.plugins.PluginManager - Disabled plugins: CSS Support, GLSL Support (1.2), HTML Tools (2.0), JavaScript Intention Power Pack (0.9.4), JavaScript Support (1.0), REST Client```

Maybe there's a way to run CLion in "verbose" mode, I'll look into this.

**Comment by Darkyenus** [comment]
Intellij 14.1 doesn't even have a getDefaultInstance() method, this is very strange. It might be caused by some antique code I haven't yet visited to clean up or by doing something wrong, which is hard to tell, since the documentation is so scarce.

**Comment by agatti** [comment]
getInstance() still works on CLion (or rather, it should), so that's not much of a concern.  However, I've tried a few things here and I wasn't able to figure out yet which logger must be put in debug mode to see why a plugin gets disabled.  I suspect it's something with the XML file, though.

**Comment by agatti** [comment]
I was able to let CLion enable (and run) the plugin with this: https://github.com/agatti/glsl4idea/commit/d659ec4c905070d2cf5d92a418ada1136d541b83 - please check if that works in IntelliJ too...

**Comment by agatti** [comment]
Oh, and by the way, I just checked with IntelliJ 14.1.1 Ultimate - FileTemplateManager.getDefaultInstance() is indeed available.

**Comment by Darkyenus** [comment]
I have CE, that might be the problem, or something is wrong with my setup. Anyway, those changes to plugin.xml didn't break anything, I have pushed them, you can try compiling it now, should be the same as your fork. Btw. how did you manage to find the solution?

**Comment by agatti** [comment]
Yep, it works, thanks a lot!  It still misses some GLES stuff but I plan to add that myself and submit a PR later on, at least now I can get it to work somehow :)

For how I fixed that, I've checked the XML file of another plugin I had installed in CLion for which I could get the source (namely https://github.com/nicoulaj/idea-markdown) - once I compared the two XML files it was fairly evident where the problem was (plugin root tag version and minimum build number for IntelliJ).

So I guess that if the reporter for #22 says it works there as well this can be closed, what do you think?

**Comment by Darkyenus** [comment]
I was hoping that you found some reference for plugin.xml (like I was looking for for some time), but this is fine as well, it doesn't look like it broke something. Thanks for your help. There is a lot of stuff still missing, I wanted to add it, but this plugin is a large beast so I wanted to understand it first. But if you manage it somehow, PRs are welcome.

This issue is by its title about building and that seems to work, so I'll close it.
(closed by https://github.com/Darkyenus/glsl4idea/commit/974ea93e772349432a50b40b908561c8ee9f6f75)
